pkg/i2gw/providers/common/converter.go

```diff
@@ -319,7 +319,23 @@ func (rg *ingressRuleGroup) configureBackendRef(paths []ingressPath) ([]gatewayv
 	var errors field.ErrorList
 	var backendRefs []gatewayv1beta1.HTTPBackendRef
 
+	// track unique backends to avoid duplicates
+	uniqueBackends := make(map[string]bool)
```

Comment:
I'd suggest using a `map[string]interface{}` and assigning `nil` to the values, as the values here do not really matter.

Reply:
Okay done: https://github.com/kubernetes-sigs/ingress2gateway/pull/104/commits/42cf5aa90af62979b2d263c2c86287041b033e4e

Reply:
Since the intended functionality here is like a set, I suggest using `map[string]struct{}` to reduce memory usage (0 vs 16 bytes per entry).